src/ge2tb.cc

```diff
@@ -293,7 +273,39 @@ void ge2tb(
                                 std::move(U_panel),
                                 std::move(TUr_panel),
                                 std::move(A_trail_j),
-                                j);
+                                j, opts2 );
+            }
+
+            // Can release tiles parallel to the main execution
+            #pragma omp task
+            {
+                for (int64_t i = k; i < A_mt; ++i) {
+                    if (A.tileIsLocal(i, k)) {
+                        A.tileUpdateOrigin(i, k);
+                        A.releaseLocalWorkspaceTile(i, k);
+                    }
+                    else {
+                        A.releaseRemoteWorkspaceTile(i, k);
+                    }
+                }
+
+                for (int64_t i : first_indices) {
+                    if (TUlocal.tileIsLocal( i, k )) {
+                        // TUlocal and TUreduce have the same process distribution
+                        TUlocal.tileUpdateOrigin( i, k );
+                        TUlocal.releaseLocalWorkspaceTile( i, k );
+                        if (i != k) {
+                            // i == k is the root of the reduction tree
+                            // TUreduce( k, k ) isn't allocated
+                            TUreduce.tileUpdateOrigin( i, k );
+                            TUreduce.releaseLocalWorkspaceTile( i, k );
+                        }
+                    }
+                    else {
+                        TUlocal.releaseRemoteWorkspaceTile( i, k );
+                        TUreduce.releaseRemoteWorkspaceTile( i, k );
+                    }
+                }
```

Comment:
For A, I'll do that (here and below).  For the `T` matrices, we can't do that because most of the tiles aren't allocated and the local releases assume the tile nodes are allocated.  Also, this loop is only `O(p)` (as in `pxq` process grid) instead of `O(mt)`.

This type of loop also appears elsewhere, so a low priority TODO would be to go through and make those all consistent.

Relatedly, I feel like we should try to fix `releaseWorkspace` for submatrices so we only need a `U_panel.releaseWorkspace()` instead of two calls (also, each loop checking the rank of each tile costs `O(0.1s)` for `mt=nt=2000`).  But, there are some things to discuss, so I'll leave that out of this PR.

Reply:
It seems there was some reasoning in having 2 calls, but I don't remember what. Kadir worked on it.

Reply:
For remote tiles, we have to check the receive count.  But, we could still make `BaseMatrix::releaseWorkspace` do
```C++
for (i = 0; i < mt; ++i) {
for (j = 0; j < nt; ++j) {
  if (tileIsLocal(i,j)) {
    releaseLocalWorkspaceTile(i, j);
  }
  else {
    releaseRemoteWorkspaceTile(i, j);
  }
}
}
```
(plus a little stuff related to releasing memory to address the current usage of `releaseWorkspace`).

Reply:
Should there be `releaseWorkspaceTile( i, j )` that checks `tileIsLocal` and does the right thing? Perhaps that should even be in MatrixStorage instead of BaseMatrix.

Reply:
Yea, that's probably the best way to do it, especially if I can remove all the existing `releaseLocalWorkspace*` and `releaseRemoteWorkspace` calls.